andela/denethor-ah-backend
# #163478819 Track comment history

#### What does this PR do?
- Implement comment edit tracking
- Implement read time algorithm

#### Description of Task to be completed?
- write tests for comment edit history
- write tests for read time algorithm
- Implement algorithm to calculate the read time
- implement fetch all articles
- implement fetch specific article
- implement update comment
- implement comment edit tracking

#### How should this be manually tested? 
- clone the repo and pull this branch `feature/163478819/track-comment-history`
- create a user using the endpoint `signup`POST `/api/users`
- verify the user using the verify link returned on signup and retrieve the token returned on successful user verification. 
- create an article and generate an articleId POST `/api/articles`
- check that the response has the field `readTime`
- To test the `comment edit history` create a comment using the endpoint 
POST `api/articles/{articleId}/comments` and retrieve the `commentId` returned from `res.body.data.id`
- update the article using the endpoint PATCH `/api/articles/{articleId}/comments/{commentId}`
- check that the response has the property, `data.oldComments` which contains the old versions of the comment.

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/163478819
https://www.pivotaltracker.com/story/show/163478810
